src/components/SignUp/OffstageRoles.tsx

```diff
@@ -1,22 +1,293 @@
+          value: 'Production Management'
+        },
+        {
+          label: 'Board Op',
+          value: 'Board Op'
+        },
+        {
+          label: 'Run Crew',
+          value: 'Run Crew'
+        }
+      ]
+    },
+    scenicAndProperties: {
+      sectionStateValue: offstageRolesScenicAndProperties,
+      sectionStateName: 'offstageRolesScenicAndProperties',
+      textHeader: 'Scenic & Properties',
+      checkboxes: [
+        {
+          label: 'Set Design',
+          value: 'Set Design'
+        },
+        {
+          label: 'Technical Direction',
+          value: 'Technical Direction'
+        },
+        {
+          label: 'Properties Designer',
+          value: 'Properties Designer'
+        },
+        {
+          label: 'Scenic Carpentry',
+          value: 'Scenic Carpentry'
+        },
+        {
+          label: 'Charge Artist',
+          value: 'Charge Artist'
+        }
+      ]
+    },
+    lighting: {
+      sectionStateValue: offstageRolesLighting,
+      sectionStateName: 'offstageRolesLighting',
+      textHeader: 'Lighting',
+      checkboxes: [
+        {
+          label: 'Lighting Design',
+          value: 'Lighting Design'
+        },
+        {
+          label: 'Projection Design',
+          value: 'Projection Design'
+        },
+        {
+          label: 'Special Effect Design',
+          value: 'Special Effect Design'
+        },
+        {
+          label: 'Master Electrician',
+          value: 'Master Electrician'
+        }
+      ]
+    },
+    sound: {
+      sectionStateValue: offstageRolesSound,
+      sectionStateName: 'offstageRolesSound',
+      textHeader: 'Sound',
+      checkboxes: [
+        {
+          label: 'Sound Design',
+          value: 'Sound Design'
+        },
+        {
+          label: 'Sound Mixer / Engineer',
+          value: 'Sound Mixer / Engineer'
+        }
+      ]
+    },
+    hairMakeupCostumes: {
+      sectionStateValue: offstageRolesHairMakeupCostumes,
+      sectionStateName: 'offstageRolesHairMakeupCostumes',
+      textHeader: 'Hair, Makeup, Costumes',
+      checkboxes: [
+        {
+          label: 'Costume Design',
+          value: 'Costume Design'
+        },
+        {
+          label: 'Hair & Wig Design',
+          value: 'Hair & Wig Design'
+        },
+        {
+          label: 'Make-up Design',
+          value: 'Make-up Design'
+        }
+      ]
+    }
+  };
+
+  const isRoleInRolesSection = (section: string[], role: string) =>
+    section.indexOf(role) > -1;
+
+  const rolesSectionChange = (
+    chkValue: any,
+    chk: string,
+    fieldName: string,
+    section: string[]
+  ) => {
+    let sectionRoles = [...section];
+    let name = fieldName;
+
+    if (chkValue) {
+      if (sectionRoles.indexOf(chk) < 0) {
+        sectionRoles.push(chk);
+      }
+    } else {
+      sectionRoles = sectionRoles.filter(rG => rG !== chk);
+    }
+
+    const target = {
+      name: 'offstageRolesGeneral',
```

Comment:
Remember, this section name needs to be dynamic, as well: you need to use `fieldName` here. Also, assigning `let name = fieldName` isn't necessary above - you can remove that line